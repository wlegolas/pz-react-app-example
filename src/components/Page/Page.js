import React from 'react';

const Page = () => (
  <div className="pz-page-container">
    <header className="pz-header">
      <h1>Header title</h1>
    </header>
    <div className="pz-page-body">
      <h1>Page body</h1>
      <div>
        <span>Content</span>
      </div>
    </div>
  </div>
);

export default Page;
