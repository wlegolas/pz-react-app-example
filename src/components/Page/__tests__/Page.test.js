import toJson from 'enzyme-to-json';
import React from 'react';
import Page from '../Page';

describe('<Page />', () => {
  let wrapper;

  beforeAll(() => {
    const pageContext = {
      userInfo: {
        id: 1,
        name: 'Legolas kun'
      }
    };

    wrapper = mount(<Page pageContext={pageContext} />);
  });

  describe('render()', () => {
    it('renders the component', () => {
      const componentJson = toJson(wrapper);

      expect(componentJson).to.matchSnapshot();
    });
  });
});
