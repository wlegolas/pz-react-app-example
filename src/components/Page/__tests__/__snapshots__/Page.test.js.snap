// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> render() renders the component 1`] = `
<Page
  pageContext={
    Object {
      "userInfo": Object {
        "id": 1,
        "name": "Legolas kun",
      },
    }
  }
>
  <div
    className="pz-page-container"
  >
    <header
      className="pz-header"
    >
      <h1>
        Header title
      </h1>
    </header>
    <div
      className="pz-page-body"
    >
      <h1>
        Page body
      </h1>
      <div>
        <span>
          Content
        </span>
      </div>
    </div>
  </div>
</Page>
`;
